Sources/Security.php

```diff
@@ -465,25 +465,22 @@ function banPermissions()
 	}
 
 	// Now that we have the mod cache taken care of lets setup a cache for the number of mod reports still open
-	if (!empty($_SESSION['rc']) && $_SESSION['rc']['time'] > $modSettings['last_mod_report_action'] && $_SESSION['rc']['id'] == $user_info['id'])
+	if (!empty($_SESSION['rc']['reports']) && !empty($_SESSION['rc']['member_reports']) && $_SESSION['rc']['time'] > $modSettings['last_mod_report_action'] && $_SESSION['rc']['id'] == $user_info['id'])
+	{
```

Comment:
That is a good point - underutilization of the cache.  isset() would be better than !empty()...   

I was avoiding a check on just ['rc'] to avoid issues like the initial bug, where ['rc'] was set (due to 'reports' being in there) so it then assumed 'member_reports' was in there when it wasn't.   Since the function is written to do one or the other, not both, we shouldn't assume to always find both.  Hope this makes sense...  

I'll test isset()...